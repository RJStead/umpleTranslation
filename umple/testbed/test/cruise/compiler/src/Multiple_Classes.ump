generate Java "../src-gen-umple";

class Class2 {
 Integer id;
 attr;
 public void Foo() {
   b = 6;
 }
 public int Bar() {
    return 0;
 }
}

class Class1
{
    0..1 -- * Class2 c2;
    public static void main(String [] argv){
	   Class1 x = new Class1();
	   Class2 c21 = new Class2(1, "1");
	   Class2 c22 = new Class2(2, "2");
	   x.addClass2(c21); x.addClass2(c22);
	   x.getClass2(1).Foo();
	   
	   String someVal = x.getClass2(0).Bar();
    }
}
