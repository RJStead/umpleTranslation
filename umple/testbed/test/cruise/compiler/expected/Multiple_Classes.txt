Multiple_Classes.ump:7: error: cannot find symbol
    b = 6;
    ^
  symbol:   variable b
  location: class Class2
Multiple_Classes.ump:24: error: incompatible types: int cannot be converted to String
       String someVal = x.getY(0).Bar();
                                     ^
2 errors